app/src/main/java/dmar/akan/akademiaandroida/features/data/remote/model/CharacterRemote.kt

```diff
@@ -0,0 +1,31 @@
+package dmar.akan.akademiaandroida.features.data.remote.model
+
+import com.google.gson.annotations.SerializedName
+import dmar.akan.akademiaandroida.features.characters.domain.model.Character
+
+data class CharacterRemote(
+    @SerializedName("id") val id: Int,
+    @SerializedName("name") val name: String,
+    @SerializedName("species") val species: String,
+    @SerializedName("type") val type: String,
+    @SerializedName("gender") val gender: String,
+    @SerializedName("origin") val origin: List<String>,
```

Comment:
To pole `origin` nie jest zwracane z API jako lista stringów, ale jest to obiekt z dwoma parametrami - `name` i `url` (świadczą o tym to tym wąsate nawiasy {}, możesz zobaczyć w dokumentacji - https://rickandmortyapi.com/documentation/#character-schema). Podobnie ma się sprawa z polem `location`. Konieczne jest to dopasować, bo nie będzie nam się ten zasób dobrze pobierał w takiej konfiguracji. 

Te dwa oddzielne obiekty możesz stworzyć w tym pliku nazywając je kolejno np. `CharacterOriginRemote` i `CharacterLocationRemote`. Następnie będzie trzeba zrobić to analogicznie w warstwie domeny oraz prezentacji.

Reply:
Dzięki za podpowiedź. Miałem dwa pomysły: 1.utworzenie inner class CharacterOriginRemote w pliku CharacterRemote, 2.utworzenie data class CharacterLocationRemote. Gdy próbowałem kompilować pojawiły się niezgodności w CharacterDisplayable :((
Kombinuję dalej.